metropolia-university-of-applied-sciences-harvard.csl

```diff
@@ -0,0 +1,440 @@
+                  <date date-parts="year" form="numeric" variable="issued"/>
+                </if>
+                <else>
+                  <text term="no date" form="short"/>
+                </else>
+              </choose>
+            </group>
+            <text variable="locator"/>
+          </group>
+        </if>
+        <else>
+          <group delimiter=", ">
+            <group delimiter=" ">
+              <text macro="title" strip-periods="false"/>
+              <choose>
+                <if variable="issued">
+                  <date variable="issued">
+                    <date-part name="year"/>
+                  </date>
+                </if>
+                <else>
+                  <text term="no date" form="short" font-style="normal"/>
+                </else>
+              </choose>
+            </group>
+            <text variable="locator"/>
+          </group>
+        </else>
+      </choose>
+    </layout>
+  </citation>
+  <bibliography>
+    <sort>
+      <key macro="contributors"/>
+      <key macro="issued-year"/>
+      <key variable="title"/>
+    </sort>
+    <layout suffix=".">
+      <choose>
+        <if variable="author editor translator" match="any">
+          <group delimiter=". ">
+            <group>
+              <text macro="contributors" suffix=" "/>
+              <text macro="issued-year" display="left-margin"/>
+            </group>
+            <text macro="title"/>
+            <text macro="description" suffix="."/>
+            <text macro="secondary-contributors" suffix="."/>
+            <group delimiter=", ">
+              <text macro="container-contributors"/>
+              <text variable="container-title"/>
+            </group>
+            <text macro="edition" suffix="."/>
+            <group delimiter=", " suffix=".">
+              <text macro="issue"/>
+              <group>
+                <text macro="locators"/>
+                <text macro="locators-article"/>
+                <text macro="locators-chapter"/>
+              </group>
+            </group>
+            <group delimiter=" ">
+              <text variable="URL" prefix="&lt;" suffix="&gt;"/>
+              <group delimiter=" " prefix="(" suffix=")">
+                <choose>
+                  <if type="post-weblog webpage" match="any">
+                    <text term="accessed" text-case="lowercase"/>
+                    <date form="numeric" variable="accessed"/>
+                  </if>
+                </choose>
+              </group>
+            </group>
+            <text macro="access"/>
+            <text variable="DOI" prefix="DOI: "/>
+            <text macro="note"/>
+          </group>
+        </if>
+        <else>
+          <group delimiter=". ">
+            <text macro="title"/>
+            <text macro="issued-year"/>
+            <choose>
+              <if type="broadcast post-weblog webpage" match="any">
+                <date form="numeric" variable="issued" prefix="(" suffix=")"/>
```

Comment:
This will duplicate the year for those tree item types -- I don't think that's right (see the auto-generated example citation)